pkg/security/ebpf/kernel/kernel.go

```diff
@@ -244,3 +249,46 @@ func (k *Version) IsAmazonLinuxKernel() bool {
 func (k *Version) IsInRangeCloseOpen(begin kernel.Version, end kernel.Version) bool {
 	return k.Code != 0 && begin <= k.Code && k.Code < end
 }
+
+func (k *Version) HaveMmapableMaps() bool {
+	if k.haveMmapableMaps != nil {
+		return *k.haveMmapableMaps
+	}
+
+	// This checks BPF_F_MMAPABLE, which appeared in 5.5 for array maps.
+	m, err := sys.MapCreate(&sys.MapCreateAttr{
+		MapType:    sys.MapType(ebpf.Array),
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: 1,
+		MapFlags:   unix.BPF_F_MMAPABLE,
+	})
+	k.haveMmapableMaps = new(bool)
+	*k.haveMmapableMaps = err == nil
+
+	if err != nil {
+		return false
+	}
+	_ = m.Close()
+	return true
+}
+
+func (k *Version) HaveRingBuffers() bool {
```

Comment:
Did you see the `cilium/ebpf` functions for this? You can do `features.HaveMapType(ebpf.RingBuf)`

Reply:
The function was not exported at the time. Will use it